### ISSUE TITLE
goimports fails to consider global variables in package

### ISSUE THREAD

**Issue by briantkennedy** [Microsoft/vscode-go#630]
I just updated to 0.6.49 and format on save is now broken for goimports.  After looking at recent changes, I believe that https://github.com/Microsoft/vscode-go/commit/b143a0d5cde4c6579a23d12f0c277370084b0975 triggered the breakage.

The difference is that the text for the file is passed via stdin instead of providing the on-disk filename.  The consequence of passing from stdin is that the goimports process is not able to determine what global variables exist within the package.  For example, if the global variable "log" is declared foo.go, while saving bar.go which calls "log.Info" or some other function which the "log" package declares, the command will operate on local data and assume you want to import the "log" package rather than using the package global variable which results in the tool incorrectly adding the "log" import.

The fix for this is to do one of the following:
1. Continue to pass the file contents via stdin and invoke as "goimports -d -srcdir [ filename ]"  (see goimports help for the -srcdir flag semantics when a filename is specified)
2. Do not pass the file contents via stdin and invoke "goimports -d [ filename ]"

Unfortunately, there doesn't appear to be a way to configure the current filename as a parameter in "go.FormatFlags" otherwise that would be an easy workaround.

**Comment by ramya-rao-a** [comment]
The reason why I changed from passing the filename to passing the filecontents via stdin was that previously formatting an unsaved file, forcefully saved the file.

#613 is an issue that `goimports` didnt find the imports for vendor packages and so removed them.
The root cause was the change mentioned above.

I put a fix in where I pass the `cwd` to the process running the formatting tool, so that it could find the vendor packages.

I'll see if that fixes your case. If not will look into your suggestions.

Regardless, I'll push an update on Monday that will contain a fix to your issue

**Comment by ramya-rao-a** [comment]
@briantkennedy I am able to repro your case.

Here is the funny thing.

I reverted the change I made regarding passing file contents via stdin instead of providing the filename to the formatting tool.

The revert fixes your issue but only when you use `goreturns`. Using `goimports` still has issues.

Can you confirm this by running both `goreturns` and `goimports` in the command line for your file?
I tried it and `goreturns` does not add the import for `log` and `goimports` does add the import.

**Comment by ramya-rao-a** [comment]
I have reverted the said change, and the latest update (0.6.50) has the fix.

**Comment by ramya-rao-a** [comment]
@briantkennedy I am working on bringing back the "pass file contents via stdin to formatter" feature.
After digging around, I found that the `srcdir` option was added by you to `goimports`. https://go-review.googlesource.com/c/tools/+/23444

But `goreturns` whose feature set is suppose to be the superset of that of `goimports` doesn't seem to have this option.

Did you have any plans to add the same option to `goreturns`?
